src/features/fireWeather/FireWeatherPage.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { selectAuthenticationReducer } from 'app/rootReducer'
+import {
+  getAuthenticationSuccess,
+  resetAuthentication,
+  getAuthenticationFailed
+} from './slices/authenticationSlice'
+
+export const FireWeatherPage = () => {
+  const { isAuthenticated } = useSelector(selectAuthenticationReducer)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const keycloak = window.Keycloak({
+      url: process.env.REACT_APP_KEYCLOAK_AUTH_URL,
+      realm: process.env.REACT_APP_KEYCLOAK_REALM,
+      clientId: process.env.REACT_APP_KEYCLOAK_CLIENT
+    })
+    keycloak
+      .init({
+        onLoad: 'login-required',
+        promiseType: 'native',
+        checkLoginIframe: false
+      })
+      .then(authenticated => {
+        if (!authenticated) {
+          dispatch(resetAuthentication())
+          window.location.reload()
+        } else {
+          dispatch(getAuthenticationSuccess(authenticated))
+        }
+        setTimeout(() => {
+          try {
+            keycloak.updateToken(60)
+          } catch (err) {
+            dispatch(getAuthenticationFailed(err))
+          }
+        }, 6000)
+      })
```

Comment:
What I was asking for was actually using Promise chaining.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises.
So right after then() block you can add .catch